validator/sawtooth_validator/journal/consensus/poet/poet_consensus/poet_block_verifier.py

```diff
@@ -0,0 +1,120 @@
+        """
+        # HACER: If we don't have any validators registered yet, we are going
+        # to immediately approve the block.  We cannot test the block's wait
+        # certificate until we can retrieve the block signer's corresponding
+        # public key.  We cannot do that until the genesis block is populated
+        # with the "initial" validator's signup information
+        validator_registry_view = ValidatorRegistryView(self._state_view)
+        if len(validator_registry_view.get_validators()) > 0:
+            try:
+                # Grab the validator info based upon the block signer's public
+                # key
+                validator_info = \
+                    validator_registry_view.get_validator_info(
+                        block_wrapper.header.signer_pubkey)
+
+                LOGGER.debug(
+                    'Block signer public key: %s',
+                    block_wrapper.header.signer_pubkey)
+                LOGGER.debug('Validator name: %s', validator_info.name)
+                LOGGER.debug('Validator id: %s', validator_info.id)
+                LOGGER.debug(
+                    'PoET public key: %s',
+                    validator_info.signup_info.poet_public_key)
+
+                # Create a list of certificates leading up to this block.
+                # This seems to have a little too much knowledge of the
+                # WaitTimer implementation, but there is no use getting more
+                # than WaitTimer.certificate_sample_length wait certificates.
+                certificates = \
+                    utils.build_certificate_list(
+                        block_header=block_wrapper.header,
+                        block_cache=self._block_cache,
+                        poet_enclave_module=self._poet_enclave_module,
+                        maximum_number=WaitTimer.certificate_sample_length)
+
+                # For the candidate block, reconstitute the wait certificate
+                # and verify that it is valid
+                wait_certificate = \
+                    utils.deserialize_wait_certificate(
+                        block=block_wrapper,
+                        poet_enclave_module=self._poet_enclave_module)
+                wait_certificate.check_valid(
+                    poet_enclave_module=self._poet_enclave_module,
+                    certificates=certificates,
+                    poet_public_key=validator_info.signup_info.poet_public_key)
+            except KeyError:
+                LOGGER.error(
+                    'Attempted to verify block from validator with no '
+                    'validator registry entry')
+                return False
+            except ValueError as ve:
+                LOGGER.error('Wait certificate is not valid: %s', str(ve))
+                LOGGER.warning('We will accept for now')
+        else:
+            LOGGER.warning(
+                'Block accepted by default because no validators registered')
```

Comment:
Yes.  This is only a phase 1 problem.  Once the genesis utility seeds with signup information we will have to make special exemption for first block, but from that point forward no special exemptions.